File: trunk/Manager/UIComposer/src/main/webapp/js/model/screen.js
/*
 * OpenRemote, the Home of the Digital Home. Copyright 2008, OpenRemote Inc.
 *
 * See the contributors.txt file in the distribution for a full listing of individual contributors.
 *
 * This is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as
 * published by the Free Software Foundation; either version 3.0 of the License, or (at your option) any later version.
 *
 * This software is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
 * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 *
 * You should have received a copy of the GNU General Public License along with this software; if not, write to the Free
 * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA, or see the FSF site:
 * http://www.fsf.org.
 */

Screen = function() {
	function Screen () {
		self = this;
		self.id = -1;
		self.name = "";
		self.label = self.name;
		//TODO add col and row
		self.row = 6;
		self.col = 4;
		
		self.buttons = new Array();
		
		self.className = getClassName(self);
	}
	
	/**
     * Create new instance from flat model (which have no private method).
     * @param model flat model (which have no private method).
     * @returns created new instance.
     */
	Screen.init = function (model) {
		var screen = new Screen();
		screen.id = model.id;
		screen.name = model.name;
		//TODO add col and row
		return screen;
	};
	
	
	return Screen;
}();

File: tags/project/UIComposer/UIComposer_1_0_0_Beta1/src/main/java/org/openremote/irbuilder/domain/Screen.java
/* OpenRemote, the Home of the Digital Home.
 * Copyright 2008, OpenRemote Inc.
 *
 * See the contributors.txt file in the distribution for a
 * full listing of individual contributors.
 *
 * This is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as
 * published by the Free Software Foundation; either version 3.0 of
 * the License, or (at your option) any later version.
 *
 * This software is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 *
 * You should have received a copy of the GNU General Public
 * License along with this software; if not, write to the Free
 * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
 * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
 */
package org.openremote.irbuilder.domain;

import com.thoughtworks.xstream.annotations.XStreamAlias;
import com.thoughtworks.xstream.annotations.XStreamAsAttribute;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author allen.wei
 */
@XStreamAlias("screen")
public class Screen extends BusinessEntity {

   @XStreamAsAttribute
   String name;
   @XStreamAsAttribute
   String icon;

   List<IPhoneButton> buttons = new ArrayList<IPhoneButton>();

   @XStreamAsAttribute
   int row;
   @XStreamAsAttribute
   int col;

   public String getName() {
      return name;
   }

   public void setName(String name) {
      this.name = name;
   }

   public String getIcon() {
      return icon;
   }

   public void setIcon(String icon) {
      this.icon = icon;
   }

   public List<IPhoneButton> getControls() {
      return buttons;
   }

   public void setControls(List<IPhoneButton> IPhoneButtons) {
      this.buttons = IPhoneButtons;
   }

   public int getRow() {
      return row;
   }

   public void setRow(int row) {
      this.row = row;
   }

   public int getCol() {
      return col;
   }

   public void setCol(int col) {
      this.col = col;
   }
}
